City-of-Helsinki/helsinki-paatokset
# PP-220:  Aggregation functionality for migrations

**To test**
* Checkout branch
* Make sure you have `PAATOKSET_OPENID_SECRET` in your .env.local file
* Add this to your .env.local: `AHJO_PROXY_BASE_URL=http://helsinki-paatokset.docker.so:8080`
* Run `make up` (needed for the new AHJO_PROXY_BASE_URL env variable), then `make shell` and `drush cim -y; drush cr;` inside the container
* Connect to VPN, then go to https://helsinki-paatokset.docker.so/fi/admin/ahjo-open-id to initiate the Open ID authentication flow
* Login with account `helfi`, ask Juho for the password
* Inside the container, run the following commands:
  * `drush ahjo-proxy:store-static-files -v` 
  * `drush ahjo-proxy:aggregate meetings --dataset=latest -v` (Some of these might fail)
  * `drush ahjo-proxy:aggregate cases --dataset=latest -v` (This might take a while)
* After this is complete, run this command to see all the Ahjo API migrations:
  * `drush migrate-status --tag=ahjo_api`
  * There should be migration derivatives for all, latest and single for cases and meetings.
* Run the "All" migration types:
  * `drush mim ahjo_meetings:all`
  * `drush mim ahjo_cases:all`
* Test that the meetings and cases content work as they did previously. Note that the "all" meetings are from HYTE, so the data quality might not be as good as production.
* Read code changes. Some parts will need to be refactored later anyway, but see if you can spot any obviously bad parts or errors
